validation/OpenSpaceToolkit/Astrodynamics/Parser.test.cpp

```diff
@@ -0,0 +1,451 @@
+    const Dictionary dataTree = Parser::ParseYaml(pathToData_, testScenario);
+
+    {
+        const SatelliteSystem satelliteSystem =
+            Parser::CreateSatelliteSystem(dataTree["success-no-prop-system"].accessDictionary());
+
+        EXPECT_EQ(satelliteSystem.getMass(), Mass::Kilograms(0.0));
+        EXPECT_EQ(satelliteSystem.getCrossSectionalSurfaceArea(), 1.0);
+        EXPECT_EQ(satelliteSystem.getDragCoefficient(), 2.2);
+        EXPECT_THROW(satelliteSystem.getPropulsionSystem(), ostk::core::error::runtime::Undefined);
+    }
+
+    {
+        const SatelliteSystem satelliteSystem =
+            Parser::CreateSatelliteSystem(dataTree["success-prop-system"].accessDictionary());
+
+        EXPECT_EQ(satelliteSystem.getMass(), Mass::Kilograms(0.0));
+        EXPECT_EQ(satelliteSystem.getCrossSectionalSurfaceArea(), 1.0);
+        EXPECT_EQ(satelliteSystem.getDragCoefficient(), 2.2);
+        EXPECT_EQ(
+            satelliteSystem.getPropulsionSystem(),
+            PropulsionSystem(
+                Scalar(0.01, PropulsionSystem::thrustSIUnit), Scalar(3000.0, PropulsionSystem::specificImpulseSIUnit)
+            )
+        );
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Validation_Parser, CreateInitialState)
+{
+    const String testScenario = {"test_parser_create_initial_state"};
+    const Dictionary dataTree = Parser::ParseYaml(pathToData_, testScenario);
+
+    VectorXd coordinates(9);
+    coordinates << Vector3d(100.0, 200.0, 300.0), Vector3d(-1.0, -2.0, -3.0), 100.0,
+        defaultSatelliteSystem_.getCrossSectionalSurfaceArea(), defaultSatelliteSystem_.getDragCoefficient();
+
+    const State defaultInitialState =
+        StateBuilder(
+            Frame::GCRF(),
+            {
+                CartesianPosition::Default(),
+                CartesianVelocity::Default(),
+                CoordinatesSubset::Mass(),
+                CoordinatesSubset::SurfaceArea(),
+                CoordinatesSubset::DragCoefficient(),
+            }
+        )
+            .build(Instant::DateTime(DateTime(2023, 1, 1, 0, 0, 0), Scale::UTC), coordinates);
+
+    {
+        const State initialState =
+            Parser::CreateInitialState(dataTree["success-cartesian"].accessDictionary(), defaultSatelliteSystem_);
+
+        EXPECT_EQ(initialState, defaultInitialState);
+    }
+
+    {
+        EXPECT_THROW(
+            Parser::CreateInitialState(dataTree["failure-keplerian"].accessDictionary(), defaultSatelliteSystem_),
+            ostk::core::error::runtime::Wrong
+        );
+    }
+
+    {
+        EXPECT_THROW(
+            Parser::CreateInitialState(dataTree["failure-time-scale"].accessDictionary(), defaultSatelliteSystem_),
+            ostk::core::error::runtime::Wrong
+        );
+    }
+
+    {
+        EXPECT_THROW(
+            Parser::CreateInitialState(dataTree["failure-frame"].accessDictionary(), defaultSatelliteSystem_),
+            ostk::core::error::runtime::Wrong
+        );
+    }
```

Comment:
What about performing the steps:
- accessDictionary()
- CreateInitialState
separately, i.e. the accessDictionary() should be done before and not be included in the EXPECT_THROW to reduce the error to be found in that step only

Also,if wanted, we can try/catch the exact error expected for the THROW

Reply:
Good point, addressed this and applied everywhere